### PR TITLE
[Plutus Core] [Pretty-printing] pad with zeroes when pretty-printing bytestring literals

### PR DESCRIPTION
**Pull request by m-alvarez**
The current prettyprinter for bytestring literals prints each 8-bit block separately as a 2-digit hex code, but doesn't add a leading zero if the block is less than 16 (10). In contrast, the pir and plc parsers expect all bytestring literals to have an even number of digits, as they're parsed two digits at a time. This causes issue #624 and a similar one in the Plutus Core parser.

### CODE REVIEW

File: language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs [hunk 1]
```diff
@@ -149,7 +149,10 @@ data Token a = LexName { loc        :: a
              deriving (Show, Eq, Generic, NFData)
 
 asBytes :: Word8 -> Doc a
-asBytes = Text 2 . T.pack . ($ mempty) . showHex
+asBytes x = Text 2 $ T.pack $ addLeadingZero $ ($ mempty) $ showHex x
```
Comment (michaelpj) [inline language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: If you're going to make it point-ful at least get rid of the weird `($ mempty)` thing.

File: language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs [hunk 2]
```diff
@@ -149,7 +149,10 @@ data Token a = LexName { loc        :: a
...
+    where addLeadingZero
```
Comment (michaelpj) [inline language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: Worth putting a type-signature on this for a bit , it took me a little while to realise how it was well-typed!

File: language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs [hunk 3]
```diff
@@ -149,7 +149,10 @@ data Token a = LexName { loc        :: a
...
+              | x < 16    = ("0" ++)
```
Comment (effectfully) [inline language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: I'd write `('0' :)`, but that's not important.

**Comment by m-alvarez** [comment]
> Test failure looks genuine - there's another one with a bytestring output.

It was! I've updated the golden file to reflect the new output (now the output is exactly the same as the one from `sha256sum`, which is probably a good thing...)

**Comment by michaelpj** [comment]
That's exciting, you got the "machine overloaded" failure on the darwin build. Annoyingly, there isn't an easy way to restart those jobs afaik, so you may just have to push a trivial change to a source file...

**Comment by michaelpj** [comment]
argh, again. How annoying.

**Comment by michaelpj** [comment]
I'm just going to force merge this, I am 100% sure it does not break only the marlowe project on darwin ;)

**Pull request merged**